prowler/providers/azure/services/search/search_service.py

```diff
@@ -0,0 +1,44 @@
+from dataclasses import dataclass
+
+from azure.mgmt.search import SearchManagementClient
+
+from prowler.lib.logger import logger
+from prowler.providers.azure.azure_provider import AzureProvider
+from prowler.providers.azure.lib.service.service import AzureService
+
+
+########################## Search
+class Search(AzureService):
+    def __init__(self, provider: AzureProvider):
+        super().__init__(SearchManagementClient, provider)
+        self.search_services = self._get_search_services()
+
+    def _get_search_services(self):
+        logger.info("Search - Getting services ...")
+        search_services = {}
+        for subscription, client in self.clients.items():
+            try:
+                search_services.update({subscription: []})
```

Comment:
```suggestion
                search_services.update({subscription: {}})
```
Please use a dict to store all the services, you can use the current ID that it is in the model as primary key for each service. Make the necessary changes in the rest of the method.